sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -172,42 +179,193 @@ class GpuGetJsonObjectMeta(
+        }
+      }
+    }
+  }
+
+  override def children: Seq[Expression] = Seq(json)
+}
+
+class GetJsonObjectCombiner(private val exp: GpuGetJsonObject) extends GpuExpressionCombiner {
+  private var id = 0
+  private val toCombine = mutable.HashMap.empty[GpuExpressionEquals, Int]
+  addExpression(exp)
+
+  override def toString: String = {
+    s"GetJsonObjCombiner $toCombine"
+  }
+
+  override def hashCode: Int = {
+    // We already know that we are GetJsonObject, and what we can combine is based
+    // on the json column being the same.
+    "GetJsonObject".hashCode + (exp.json.semanticHash() * 17)
+  }
+
+  override def equals(o: Any): Boolean = o match {
+    case other: GetJsonObjectCombiner =>
+      exp.json.semanticEquals(other.exp.json) &&
+        // We don't support multi-get with the save path for verify yet
+        exp.savePathForVerify.isEmpty &&
+        other.exp.savePathForVerify.isEmpty
+    case _ => false
+  }
+
+  override def addExpression(e: Expression): Unit = {
+    val localId = id
+    id += 1
+    val key = GpuExpressionEquals(e)
+    if (!toCombine.contains(key)) {
+      toCombine.put(key, localId)
+    }
+  }
 
+  override def useCount: Int = toCombine.size
+
+  private def fieldName(id: Int): String =
+    s"_mgjo_$id"
+
+  @scala.annotation.tailrec
+  private def extractLit(exp: Expression): Option[GpuLiteral] = exp match {
+    case l: GpuLiteral => Some(l)
+    case a: Alias => extractLit(a.child)
+    case _ => None
+  }
+
+  private lazy val multiGet: GpuMultiGetJsonObject = {
+    val json = toCombine.head._1.e.asInstanceOf[GpuGetJsonObject].json
+    val fieldsNPaths = toCombine.toSeq.map {
+      case (k, id) =>
+        (id, k.e)
+    }.sortBy(_._1).map {
+      case (id, e: GpuGetJsonObject) =>
+        val parsedPath = extractLit(e.path).flatMap { s =>
+          import GpuGetJsonObject._
+          val str = s.value match {
+            case u: UTF8String => u.toString
+            case _ => null.asInstanceOf[String]
+          }
+          val pathInstructions = parseJsonPath(str)
+          if (hasSeparateWildcard(pathInstructions)) {
+            // If has separate wildcard path, should return all nulls
+            None
+          } else {
+            // Filter out the unneeded instructions before we cache it
+            pathInstructions.map(JsonPathParser.filterInstructionsForJni)
+          }
+        }
+        (StructField(fieldName(id), e.dataType, e.nullable), parsedPath)
+    }
+    val dt = StructType(fieldsNPaths.map(_._1))
+    val conf = SQLConf.get
+    val targetBatchSize = RapidsConf.GPU_BATCH_SIZE_BYTES.get(conf)
+    val tmp = conf.getConfString("spark.sql.test.multiget.parallel", null)
+    val parallel = Option(tmp).map(_.toInt)
+    GpuMultiGetJsonObject(json, fieldsNPaths.map(_._2), dt)(targetBatchSize, parallel)
+  }
+
+  override def getReplacementExpression(e: Expression): Expression = {
+    val localId = toCombine(GpuExpressionEquals(e))
```

Comment:
Given the `addExpression` logic and this, I believe this is deduping, so we continue to get the same field from the struct, since `localId` will be the smallest id at the time when the first equal expression (in the set) was added. Just making sure I understand

Reply:
Yes that is correct